docs/specs/priority-queue.md

```diff
@@ -124,46 +129,53 @@ Then simply call the default constructor (`Comparer<T>.Default` is assumed):
 var queue = new PriorityQueue<MyClass>();
 ```
 
-#### (4)
-
-Priority for `MyClass` is obtainable from some other objects, for example a
-dictionary. It is done analogically to **(2)**, simply by some custom logic in
-the comparer.
-
 ## `PriorityQueue<TElement, TPriority>`
 
 ```csharp
 public class PriorityQueue<TElement, TPriority> :
-    IEnumerable,
-    IEnumerable<(TElement element, TPriority priority)>,
-    IReadOnlyCollection<(TElement element, TPriority priority)>
+    IDictionary<TElement, TPriority>,
```

Comment:
What happens when two elements that are equal end up in the queue?

Reply:
@KrzysztofCwalina Hm. I thought I understood this, but expanding it out I realize I need to make a tweak. The guideline I think we should follow here is that the classes should follow the contracted behavior for `.Add()` of the interfaces they implement. Specifically `PriorityQueue<T>` should just 'noop' (at least I believe that is what ISet<T>.Add() does - or does it 'replace' the element with itself?) and return false to indicate the element was already present. I need to fix the return type of `PriorityQueue<TElement,TPriority>.Add()` to void to match `IDictionary<TElement,TPriotity>`and it should throw `ArgumentException`.